### PR TITLE
Add notifcation index view

### PR DESCRIPTION
**Pull request by CouchofTomato**
Because: We want users to be able to view their notifications

This commit:
- Remove notification argument to NotifyUser as it's redundant
- Update method to DeadLink url to return correct lesson path
- Add Notification ViewComponent
- Add helper method to render svgs

### CODE REVIEW

**Comment by CouchofTomato** [comment]
odin-notifications

**Comment by codyloyd** [comment]
design and styles LGTM.  I'm too far out of the loop with Ruby/Rails to comment on the rest tho.

File: app/components/notification_component.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# rubocop:disable Lint/MissingSuper
```
Comment (KevinMulhern) [inline app/components/notification_component.rb:3]: Haven't seen this one before 😄 does it not like the initializer?

File: spec/components/notification_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+           id: 1,
+           title: 'test title',
+           message: 'test message')
+  end
+
+  before do
+    render_inline(described_class.new(notification: notification))
+  end
+
+  it 'renders the component with a link to the notification\'s url' do
```
Comment (KevinMulhern) [inline spec/components/notification_component_spec.rb:15]: These tests are 🔥

**Review by KevinMulhern — APPROVED** [review]
This looks brilliant @CouchofTomato, awesome work mate!

**Pull request merged**